packages/frontend-shared/src/gql-components/HeaderBar.vue

```diff
@@ -36,8 +37,14 @@ const props = withDefaults(
 
 const emit = defineEmits<{
   (event: 'connect-project'): void
+  (event: 'header-loaded'): void
 }>()
 
 const query = useQuery({ query: HeaderBar_HeaderBarQueryDocument })
 
+watchEffect(() => {
+  if (query.data.value) {
+    emit('header-loaded')
+  }
+})
```

Comment:
Instead of emitting an event out and then using that value to decide to show the header, this logic can just stay encapsulated in this component.  Just move the `v-if="query.data.value"` on the `HeaderBarContent` component above to the wrapping `div` and it will handle only showing the banner slot once the header GQL query has completed.